05/app.js

```diff
@@ -1,37 +1,69 @@
 const apiUrl = 'http://localhost:3000/users';
 
+const submitBtn = document.querySelector('.form__submit');
 document.addEventListener('DOMContentLoaded', init);
 
 function init() {
-    loadUsers();
+	loadUsers();
+	submitBtn.addEventListener('click', getDataFromInput);
 }
 
 function loadUsers() {
-    const promise = fetchGet(apiUrl);
+	const promise = fetchGet(apiUrl);
 
-    promise
-        .then(data => insertUsers(data))
-        .catch(err => console.error(err));
+	promise.then(data => insertUsers(data)).catch(err => console.error(err));
 }
 
 function fetchGet(url) {
-    return fetch(url)
-        .then(resp => {
-            if(resp.ok) {
-                return resp.json();
-            }
-
-            return Promise.reject(resp);
-        });
+	return fetch(url).then(resp => {
+		if (resp.ok) {
+			return resp.json();
+		}
+
+		return Promise.reject(resp);
+	});
+}
+
+function getDataFromInput(e) {
+	e.preventDefault();
+	const name = document.querySelector('.form__field--first-name').value;
+	const lastName = document.querySelector('.form__field--last-name').value;
+
+	if (name != '' && lastName != '') {
+		const userData = {
+			firstName: name,
+			lastName: lastName,
+		};
+		sendDataToServer(userData);
+	} 
+}
+function sendDataToServer(userData) {
+	const options = {
+		method: 'POST',
+		body: JSON.stringify(userData),
+		headers: { 'Content-Type': 'application/json' },
+	};
+	const promise = fetch(apiUrl, options);
+
+	promise
+		.then(resp => {
+			if (resp.ok) {
+				return resp.json();
+			}
+			return Promise.reject(resp);
+		})
+		.then(data => console.log(data))
+		.catch(err => console.error(err));
+        // nie dodaje finally() z odswiezaniem wynikow, bo widze ze nie jest to potrzebne 
```

Comment:
ok, dla pewności warto to zrobić. To działania live-server-a powoduje, że nie trzeba odświeżać, ale w realnym projekcie będzie trzeba ;)